### PR TITLE
[Cleanup]: Fix Android Studio Warnings #13282

### PR DESCRIPTION
**Pull request by eliseevavika**
## Purpose / Description
The purpose of this pull request is to address task #13282, titled '[Cleanup]: Fix Android Studio Warnings,' which is designed to assist in onboarding new users to the codebase by resolving various warnings and errors in the code. 
Upon inspecting the codebase, I identified several errors, warnings, and weak warnings.
To familiarize myself with the codebase and practice my skills, I have fixed a subset of these issues. This effort aims to improve the overall code quality and maintainability of the project

## Fixes
* Fixed some the errors, warnings + weak warnings in the file after code Inspection (Code -> Inspect Code)

## How Has This Been Tested?

I have tested the changes by running the application on my device. The project launched successfully without any errors or issues, confirming the functionality of the implemented features

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt [hunk 1]
```diff
@@ -331,7 +331,7 @@ open class AnkiActivity : AppCompatActivity, SimpleMessageDialogListener {
             .setToolbarColor(toolbarColor)
             .setNavigationBarColor(navBarColor)
             .build()
-        val builder = CustomTabsIntent.Builder(customTabActivityHelper.session)
+        val builder = Builder(customTabActivityHelper.session)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt:334]: Better as it was previously

If it's not possible to resolve this another way, could you alias `CustomTabsIntent.Builder` to `CustomTabsIntentBuilder`

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Scheduler.kt [hunk 2]
```diff
@@ -590,16 +590,16 @@ open class Scheduler(val col: Collection) {
 
         // Every queue has a failure rate, and each failure will become a relrn
         var toRelrn = counts.new // Assume every new card becomes 1 relrn
-        toRelrn += Math.ceil((1 - relrnRate) * counts.lrn).toInt()
-        toRelrn += Math.ceil((1 - revRate) * counts.rev).toInt()
+        toRelrn += kotlin.math.ceil((1 - relrnRate) * counts.lrn).toInt()
+        toRelrn += kotlin.math.ceil((1 - revRate) * counts.rev).toInt()
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/Scheduler.kt:594]: is `ceil` usable?

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Scheduler.kt [hunk 3]
```diff
@@ -590,16 +590,16 @@ open class Scheduler(val col: Collection) {
...
 
         // Use the accuracy rate of the relrn queue to estimate how many reps we will end up with if the cards
         // currently in relrn continue to fail at that rate. Loop through the failures of the failures until we end up
         // with no predicted failures left.
 
         // Cap the lower end of the success rate to ensure the loop ends (it could be 0 if no revlog history, or
         // negative for other reasons). 5% seems reasonable to ensure the loop doesn't iterate too much.
-        relrnRate = Math.max(relrnRate, 0.05)
+        relrnRate = kotlin.math.max(relrnRate, 0.05)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sched/Scheduler.kt:602]: ditto: shouldn't need the `kotlin.math` qualifier

File: annotations/README.md [hunk 4]
```diff
@@ -40,6 +40,10 @@ Sample:
           <root type="composite">
             <root url="jar://$USER_HOME$/AppData/Local/Android/Sdk/platforms/android-28/data/annotations.zip!/" type="simple" />
             <root url="file://C:/GitHub/Anki-Android-David/annotations" type="simple" />
+          </root>
```
Comment (david-allison) [inline annotations/README.md:43]: This doesn't seem to add value to the example

File: api/src/main/java/com/ichi2/anki/api/AddContentApi.kt [hunk 5]
```diff
@@ -83,11 +83,16 @@ public class AddContentApi(context: Context) {
         val cardsUri = Uri.withAppendedPath(newNoteUri, "cards")
         val cardsQuery = resolver.query(cardsUri, null, null, null, null) ?: return null
         cardsQuery.use { cardsCursor ->
-            while (cardsCursor.moveToNext()) {
-                val ord = cardsCursor.getString(cardsCursor.getColumnIndex(Card.CARD_ORD))
-                val cardValues = ContentValues().apply { put(Card.DECK_ID, deckId) }
-                val cardUri = Uri.withAppendedPath(Uri.withAppendedPath(newNoteUri, "cards"), ord)
-                resolver.update(cardUri, cardValues, null, null)
+            val ordColumnIndex = cardsCursor.getColumnIndex(Card.CARD_ORD)
```
Comment (david-allison) [inline api/src/main/java/com/ichi2/anki/api/AddContentApi.kt:86]: I didn't see an IDE warning for this one

File: api/src/main/java/com/ichi2/anki/api/AddContentApi.kt [hunk 6]
```diff
@@ -284,19 +289,27 @@ public class AddContentApi(context: Context) {
...
             while (cardsCursor.moveToNext()) {
-                // add question and answer for each card to map
-                val n = cardsCursor.getString(cardsCursor.getColumnIndex(Card.CARD_NAME))
-                val q = cardsCursor.getString(cardsCursor.getColumnIndex(Card.QUESTION))
-                val a = cardsCursor.getString(cardsCursor.getColumnIndex(Card.ANSWER))
-                cards[n] = hashMapOf(
-                    "q" to q,
-                    "a" to a
-                )
+                val nameColumnIndex = cardsCursor.getColumnIndex(Card.CARD_NAME)
```
Comment (david-allison) [inline api/src/main/java/com/ichi2/anki/api/AddContentApi.kt:292]: ditto, which warnings are fixed

**Review by david-allison — CHANGES_REQUESTED** [review]
Thanks! Small amount of feedback

Please revert all changes to `AddContentApi` so we can get this merged as soon as possible

**Comment by david-allison** [comment]
```
* What went wrong:
Execution failed for task ':AnkiDroid:ktlintMainSourceSetCheck'.
> A failure occurred while executing org.jlleitschuh.gradle.ktlint.worker.ConsoleReportWorkAction
   > KtLint found code style violations. Please see the following reports:
     - /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/build/reports/ktlint/ktlintMainSourceSetCheck/ktlintMainSourceSetCheck.txt
```

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Scheduler.kt (hunk 2 above)
Comment (eliseevavika) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/Scheduler.kt:594]: In Java, we use Math.ceil(), while in Kotlin, we use kotlin.math.ceil(), which is to round up the result to the nearest integer. Both functions perform the same operation, rounding up to the nearest integer.

File: annotations/README.md (hunk 4 above)
Comment (eliseevavika) [reply annotations/README.md:43]: there was a warning or error at the end of the file. Should I undo these changes?

**Comment by Aditya13s** [comment]
> Task :AnkiDroid:ktlintMainSourceSetCheck FAILED
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt:30:1 Unused import

@eliseevavika There is an unused import in AnkiActivity.kt in line 30. Remove it

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt (hunk 1 above)
Comment (eliseevavika) [reply AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt]: reverted changes to AddContentApi

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Scheduler.kt (hunk 3 above)
Comment (eliseevavika) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/Scheduler.kt]: made changes: used import statement for max function in Scheduler.kt and removed kotlin.math

**Comment by eliseevavika** [comment]
> > Task :AnkiDroid:ktlintMainSourceSetCheck FAILED
> > /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt:30:1 Unused import
> 
> @eliseevavika There is an unused import in AnkiActivity.kt in line 30. Remove it

by a comment above I reverted the change for CustomTabsIntent, so now I need this import:
import androidx.browser.customtabs.CustomTabsIntent

File: annotations/README.md (hunk 4 above)
Comment (david-allison) [reply annotations/README.md:43]: Yep - the closing tags don't seem to provide value for readers

File: api/src/main/java/com/ichi2/anki/api/AddContentApi.kt (hunk 5 above)
Comment (eliseevavika) [reply api/src/main/java/com/ichi2/anki/api/AddContentApi.kt:86]: I do the next steps:
Code - > Inspect Code (Whole project)
Under the folder -Android Lint: Correctness:
there are 12 warnings in the file AddContentApi.kt:
"value must be >=0 but ‘getColumnIndex’ can be -1

Screen Shot 2024-03-31 at 3 15 25 PM
"

File: annotations/README.md (hunk 4 above)
Comment (eliseevavika) [reply annotations/README.md]: undo my changes

**Comment by eliseevavika** [comment]
> ```
> * What went wrong:
> Execution failed for task ':AnkiDroid:ktlintMainSourceSetCheck'.
> > A failure occurred while executing org.jlleitschuh.gradle.ktlint.worker.ConsoleReportWorkAction
>    > KtLint found code style violations. Please see the following reports:
>      - /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/build/reports/ktlint/ktlintMainSourceSetCheck/ktlintMainSourceSetCheck.txt
> ```

I see that all (7) checks have passed. How can I see this error?

File: AnkiDroid/src/main/java/com/ichi2/libanki/sched/Scheduler.kt (hunk 2 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sched/Scheduler.kt:595]: Yes, but the namespace shouldn't be necessary: `ceil` vs `kotlin.math.ceil`

File: api/src/main/java/com/ichi2/anki/api/AddContentApi.kt (hunk 5 above)
Comment (david-allison) [reply api/src/main/java/com/ichi2/anki/api/AddContentApi.kt:86]: Revert the file for now, we can do better when cleaning this up

**Review by david-allison — CHANGES_REQUESTED** [review]
One request for a revert on `AddContentApi`

Optional nitpick with `kotlin.math.ceil`

Then good to merge, thanks!

**Review by david-allison — APPROVED** [review]
Looks great, thanks!

**Review by lukstbit — APPROVED** [review]
LGTM, thank you for contributing!

**Comment by mikehardy** [comment]
great! merge queue was drained, so squashed this in

**Pull request merged**